### DISCUSSION TITLE
Async Unittests

### DISCUSSION THREAD

**Discussion by rednaks** [rednaks/django-async-orm#13] [General]
I've been working on making tests run in async mode, thankfully I found that it's already supported using the `IsolatedAsyncioTestCase` but still it fails in some cases and can't understand why.

The failing test is `test_delete` https://github.com/rednaks/django-async-orm/blob/main/tests/test_django_async_orm.py#L111
I create an object, and then delete everything.
but before deleting I wanted to test if the object really exists in the database. so get retrieve all objects : `TestModel.objects.async_all()`
but here where things get weird, the list is empty ..

I tried to understand why, here is what I investigated:
1. Maybe the tasks are executed in different threads: 

 No `sync_to_async` is `thread_sensitive=True` for all patched `QuerySetAsync` methods, unless I'm missing something, this should not be the case.

2.  `IsolatedAsyncioTestCase` is using different loops ?

The source code if `unittests` shows that it's the same loop for all tests. :  https://github.com/python/cpython/blob/main/Lib/unittest/async_case.py#L159

3. `async_create` is not persisting the object ? : 

No because django doc says that `create` is equivalient to instanciating object and `.save()`

4. Different Manager for each task ? : 

No, `id(TestModel.objects)` is the same.

**Comment by rednaks** [comment]
any ideas @allen-munsch ?

**Comment by rednaks** [comment]
I'm a bit confused now, becaue even in sync orm I have the same issue
```python
class SyncTestCase(TestCase):

    def test_delete(self):
        created = TestModel.objects.create(name='sync to delete')
        all_created = TestModel.objects.all()
        print(all_created) 

```

output

```
<QuerySetAsync []>
```

**Comment by rednaks** [comment]
Well, the whole issue was about that I was using `TestCase` instead of `TransactionTestCase` ..

**Reply by allen-munsch** [reply]
Ah, yeah that makes sense.

Just saw this. Glad you were able to find it.

I've been bitten by that one in the past. Probably would today too. haha.

**Reply by rednaks** [reply]
yeah I spent some time looking for if, I totally forgot I should use that class for db related tests. 

Thank you anyways
